syncopy/tests/test_specest_fooof.py

```diff
@@ -0,0 +1,184 @@
+            will not off by one. Also it is questionable whether users would want that. We therefore use
+            consider it an error to pass an input frequency axis that contains the zero, and throw an
+            error in the frontend to stop before any expensive computations happen. This test checks for
+            that error.
+        """
+        self.cfg['output'] = "fooof"
+        self.cfg['foilim'] = [0., 250.]    # Include the zero in tfData.
+        self.cfg['out'] = None
+        with pytest.raises(SPYValueError) as err:
+            _ = freqanalysis(self.cfg, self.tfData)  # tfData contains zero.
+        assert "a frequency range that does not include zero" in str(err.value)
+
+    def test_fooof_output_fooof_works_with_freq_zero_in_data_after_setting_foilim(self, fulltests):
+        """
+        This tests the intended operation with output type 'fooof': with an input that does not
+        include zero, ensured by using the 'foilim' argument/setting when calling freqanalysis.
+
+        This returns the full, fooofed spectrum.
+        """
+        self.cfg['output'] = "fooof"
+        self.cfg['foilim'] = [0.5, 250.]    # Exclude the zero in tfData.
+        self.cfg['out'] = None
+        spec_dt = freqanalysis(self.cfg, self.tfData)
+
+        # check frequency axis
+        assert spec_dt.freq.size == 500
+        assert spec_dt.freq[0] == 0.5
+        assert spec_dt.freq[499] == 250.
+
+        # check the log
+        assert "fooof_method = fooof" in spec_dt._log
+        assert "fooof_aperiodic" not in spec_dt._log
+        assert "fooof_peaks" not in spec_dt._log
+        assert "fooof_opt" in spec_dt._log
+
+        # check the data
+        assert spec_dt.data.ndim == 4
+        assert spec_dt.data.shape == (1, 1, 500, 1)
+        assert not np.isnan(spec_dt.data).any()
+
+        # check that the cfg is correct (required for replay)
+        assert spec_dt.cfg['freqanalysis']['output'] == 'fooof'
+
+        # Plot it.
+        #  _plot_powerspec(freqs=spec_dt.freq, powers=spec_dt.data[0, 0, :, 0])
+        #spec_dt.singlepanelplot()
+
+    def test_spfooof_output_fooof_aperiodic(self, fulltests):
+        """Test fooof with output type 'fooof_aperiodic'. A spectrum containing only the aperiodic part is returned."""
+        self.cfg['output'] = "fooof_aperiodic"
+        self.cfg['foilim'] == [0.5, 250.]
+        self.cfg['out'] = None
+        spec_dt = freqanalysis(self.cfg, self.tfData)
+
+        # log
+        assert "fooof" in spec_dt._log  # from the method
+        assert "fooof_method = fooof_aperiodic" in spec_dt._log
+        assert "fooof_peaks" not in spec_dt._log
+
+        # check the data
+        assert spec_dt.data.ndim == 4
+        assert spec_dt.data.shape == (1, 1, 500, 1)
+        assert not np.isnan(spec_dt.data).any()
+        _plot_powerspec(freqs=spec_dt.freq, powers=np.ravel(spec_dt.data))
+
+    def test_spfooof_output_fooof_peaks(self, fulltests):
```

Comment:
This is the result, and it's not very pretty.. :sweat_smile:  That connects to my commment above to use another synthetic data generating procedure
![Screenshot from 2022-07-18 16:16:52](https://user-images.githubusercontent.com/10595420/179531627-12382464-13c1-4826-b375-5fc71305734d.png)